src/CommunityToolkit.Maui/Categories/IScreenCategorizer.shared.cs

```diff
@@ -0,0 +1,13 @@
+﻿
+using CommunityToolkit.Maui.Core.Primitives;
+using CommunityToolkit.Maui.Mappings;
+
+namespace  CommunityToolkit.Maui.Categories;
+
+/// <summary>
+/// Responsible for identify the screen size of the device model on either device name or device screen dimensions.
+/// </summary>
+ interface IScreenCategorizer
+{
+	ScreenCategories GetCategoryByDiagonalSize(List<SizeMappingInfo> mappings, double deviceActualDiagonalSize);
```

Comment:
Looking into the implementation we don't use any feature from `List`, so how about changing this to be an `IEnumerable`?

```suggestion
	ScreenCategories GetCategoryByDiagonalSize(IEnumerable<SizeMappingInfo> mappings, double deviceActualDiagonalSize);
```